app/models/setting/email.rb

```diff
@@ -1,5 +1,12 @@
+require 'shellwords'
+
 class Setting::Email < Setting
   NON_EMAIL_YAML_SETTINGS = %w(send_welcome_email email_reply_address email_subject_prefix)
+  SENDMAIL_LOCATIONS = %w(/usr/sbin/sendmail /usr/bin/sendmail /usr/local/sbin/sendmail /usr/local/bin/sendmail)
+
+  def self.sendmail_locations_hash
+    SENDMAIL_LOCATIONS.zip(SENDMAIL_LOCATIONS).to_h
+  end
```

Comment:
I didn't find such an example, but looks like we can also return an array from the proc: https://github.com/theforeman/foreman/blob/3256e1396e157b8a84960fc40e4754ab0ebc9c98/app/services/setting_value_selection.rb#L7

Reply:
I am getting no implicit conversion of Symbol into Integer when I pass array.

```
app/services/setting_value_selection.rb:25:in `[]'
app/services/setting_value_selection.rb:25:in `block in editable_select_optgroup'
app/services/setting_value_selection.rb:24:in `each'
app/services/setting_value_selection.rb:24:in `reduce'
app/services/setting_value_selection.rb:24:in `editable_select_optgroup'
app/services/setting_value_selection.rb:8:in `initialize'
app/services/setting_select_collection.rb:18:in `new'
app/services/setting_select_collection.rb:18:in `collection_for'
app/presenters/setting_presenter.rb:74:in `select_values'
app/helpers/settings_helper.rb:17:in `setting_to_hash'
app/helpers/settings_helper.rb:3:in `block in grouped_settings'
app/services/setting_registry.rb:16:in `each'
app/services/setting_registry.rb:16:in `each'
app/helpers/settings_helper.rb:3:in `map'
```

I am keeping it as a hash.